rclcpp/include/rclcpp/client.hpp

```diff
@@ -44,15 +46,23 @@ class ClientBase
 public:
   RCLCPP_MAKE_SHARED_DEFINITIONS(ClientBase);
 
-  ClientBase(rmw_client_t * client_handle, const std::string & service_name)
-  : client_handle_(client_handle), service_name_(service_name)
+  ClientBase(
+    std::shared_ptr<rmw_node_t> node_handle,
+    rmw_client_t * client_handle,
+    const std::string & service_name)
+  : node_handle_(node_handle), client_handle_(client_handle), service_name_(service_name)
   {}
 
   ~ClientBase()
```

Comment:
Unrelated to the PR itself but probably necessary for cleaning up user created instances:

Shouldn't the destructor be virtual?